### PR TITLE
[android] missing DeleteLocalRef

### LINKED ISSUE #11212: [Android] [6.0.0-beta.2] local reference table overflow (max=512)
Platform: Android
Version: 6.0.0-beta.2

I've just tested the new beta version of the Mapbox Android SDK.

In our app we draw routes based on polygons. Since the last update, we get following native crash:

```
art/runtime/indirect_reference_table.cc:115] JNI ERROR (app bug): local reference table overflow (max=512)
art/runtime/indirect_reference_table.cc:115] local reference table dump:
art/runtime/indirect_reference_table.cc:115]   Last 10 entries (of 509):
art/runtime/indirect_reference_table.cc:115]       508: 0x133d4df0 java.lang.Double
art/runtime/indirect_reference_table.cc:115]       507: 0x133d4dd0 java.lang.Double
art/runtime/indirect_reference_table.cc:115]       506: 0x133d4d80 java.lang.Double
art/runtime/indirect_reference_table.cc:115]       505: 0x133d4d70 java.lang.Double
art/runtime/indirect_reference_table.cc:115]       504: 0x133d4d60 java.lang.Double
art/runtime/indirect_reference_table.cc:115]       503: 0x133d4d50 java.lang.Double
art/runtime/indirect_reference_table.cc:115]       502: 0x133d4d40 java.lang.Double
art/runtime/indirect_reference_table.cc:115]       501: 0x133d4d30 java.lang.Double
art/runtime/indirect_reference_table.cc:115]       500: 0x133d4d20 java.lang.Double
art/runtime/indirect_reference_table.cc:115]       499: 0x133d4d10 java.lang.Double
art/runtime/indirect_reference_table.cc:115]   Summary:
art/runtime/indirect_reference_table.cc:115]         1 of com.google.gson.JsonObject
art/runtime/indirect_reference_table.cc:115]         1 of com.mapbox.geojson.AutoValue_Polygon
art/runtime/indirect_reference_table.cc:115]         1 of com.mapbox.geojson.AutoValue_Feature
art/runtime/indirect_reference_table.cc:115]         2 of java.lang.Object[] (1 elements) (2 unique instan
art/runtime/indirect_reference_table.cc:115]         1 of java.lang.Object[] (263 elements)
art/runtime/indirect_reference_table.cc:115]         2 of java.lang.Class (2 unique instances)
art/runtime/indirect_reference_table.cc:115]         1 of java.lang.String[] (3 elements)
art/runtime/indirect_reference_table.cc:115]         3 of java.lang.String (3 unique instances)
art/runtime/indirect_reference_table.cc:115]       494 of java.lang.Double (494 unique instances)
art/runtime/indirect_reference_table.cc:115]         3 of java.util.ArrayList (3 unique instances)
```

The culprit here seems to be the massive amount of Points we use in a Feature (Polygon).

The worked flawlessly in version 5.4.0.

### PR DESCRIPTION
**Pull request by osana**
This should resolve #11212 but should be tested before it could be closed

### CODE REVIEW

File: platform/android/src/geojson/polygon.cpp [hunk 1]
```diff
@@ -29,6 +29,7 @@ mapbox::geojson::polygon Polygon::convert(jni::JNIEnv &env, jni::Object<java::ut
         for (auto&& line : multiLine) {
             polygon.emplace_back(convertExplicit<mapbox::geojson::linear_ring>(std::move(line)));
         }
+        jni::DeleteLocalRef(env, multiLine);
```
Comment (tobrun) [inline platform/android/src/geojson/polygon.cpp]: I think `multiLine` should be `jPointListsList`

Comment (osana) [reply platform/android/src/geojson/polygon.cpp]: @tobrun `jPointListsList` is a parameter in that function so caller will DeleteLocalRef to it

Comment (tobrun) [reply platform/android/src/geojson/polygon.cpp]: yes 🤦‍♂️

File: platform/android/src/geojson/feature_collection.cpp [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#include <android/log.h>
```
Comment (tobrun) [inline platform/android/src/geojson/feature_collection.cpp]: log can be removed

File: platform/android/src/geojson/point.cpp [hunk 3]
```diff
@@ -1,6 +1,8 @@
+#include <android/log.h>
```
Comment (tobrun) [inline platform/android/src/geojson/point.cpp]: log can be removed

**Comment by osana** [comment]
cherry-picked into `release-boba` : https://github.com/mapbox/mapbox-gl-native/pull/11243

**Pull request merged**